DataAccessTest::testFetchColumnNonExistingIndex() covers a PDO bug

According to the discussion in https://github.com/php/php-src/pull/3655, the fact that `PDOStatement::fetchColumn()` returns `null` for an invalid column index, is a bug. Starting 7.2.14 and other newer PHP versions, `DataAccessTest::testFetchColumnNonExistingIndex()` will fail:
```
↪ php --version
PHP 7.2.14-dev (cli) (built: Nov 26 2018 15:14:02) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies

↪ phpunit --filter testFetchColumnNonExistingIndex tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php
PHPUnit 7.4.0 by Sebastian Bergmann and contributors.

Testing Doctrine\Tests\DBAL\Functional\DataAccessTest
E                                                                                   1 / 1 (100%)

Time: 39 ms, Memory: 6.00MB

There was 1 error:

1) Doctrine\Tests\DBAL\Functional\DataAccessTest::testFetchColumnNonExistingIndex
Exception: [Doctrine\DBAL\Driver\PDOException] SQLSTATE[HY000]: General error: Invalid column index

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
```
This test was introduced as part of https://github.com/doctrine/dbal/pull/709 but it doesn't cover the issue which was fixed.

Instead of supporting this behavior by means of a workaround, I think we should just remove the test and document the changes. Invalid column index usually is a sign of a logical error in the client code and shouldn't be silently suppressed by the DBAL.

--- a/UPGRADE.md
+++ b/UPGRADE.md
@@ -1,5 +1,9 @@
 # Upgrade to 2.9
 
+## Deprecated `Statement::fetchColumn()` with an invalid index
+
+Calls to `Statement::fetchColumn()` with an invalid column index currently return `NULL`. In the future, such calls will result in a exception.
+
 ## Deprecated `Configuration::getFilterSchemaAssetsExpression()`, `::setFilterSchemaAssetsExpression()` and `AbstractSchemaManager::getFilterSchemaAssetsExpression()`.
 
 Regular expression-based filters are hard to extend by combining together. Instead, you may use callback-based filers via `::getSchemaAssetsFilter()` and `::getSchemaAssetsFilter()`. Callbacks can use regular expressions internally.

--- a/tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php
+++ b/tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php
@@ -891,22 +891,6 @@ class DataAccessTest extends DbalFunctionalTestCase
     /**
      * @group DBAL-1028
      */
-    public function testFetchColumnNonExistingIndex()
-    {
-        if ($this->connection->getDriver()->getName() === 'pdo_sqlsrv') {
-            $this->markTestSkipped(
-                'Test does not work for pdo_sqlsrv driver as it throws a fatal error for a non-existing column index.'
-            );
-        }
-
-        self::assertNull(
-            $this->connection->fetchColumn('SELECT test_int FROM fetch_table WHERE test_int = ?', [1], 1)
-        );
-    }
-
-    /**
-     * @group DBAL-1028
-     */
     public function testFetchColumnNoResult()
     {
         self::assertFalse(

